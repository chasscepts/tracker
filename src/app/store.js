import { configureStore } from '@reduxjs/toolkit';
import authReducer from '../reducers/authSlice';
import tasksReducer from '../reducers/tasksSlice';
import timerReducer from '../reducers/timerSlice';

export default configureStore({
  reducer: {
    auth: authReducer,
    tasks: tasksReducer,
    timer: timerReducer,
  },
});
