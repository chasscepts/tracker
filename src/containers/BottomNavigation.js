import { useLocation } from 'react-router';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import home from '../assets/images/home.png';
import plus from '../assets/images/plus.png';
import editTask from '../assets/images/edit-task.png';
import chart from '../assets/images/chart.png';
import styles from '../assets/css/BottomNavigation.module.css';
import { selectUser } from '../reducers/authSlice';

function Item({
  icon, path, label, matches,
}) {
  const { pathname } = useLocation();
  let navClass = styles.nav;
  if (pathname === path || matches.indexOf(pathname) >= 0) {
    navClass = `${navClass} ${styles.active}`;
  }

  return (
    <Link to={path}>
      <div className={navClass}>
        <img src={icon} alt={path} />
        <div className={styles.label}>{label}</div>
      </div>
    </Link>
  );
}

Item.propTypes = {
  icon: PropTypes.node.isRequired,
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  matches: PropTypes.arrayOf(PropTypes.string),
};

Item.defaultProps = {
  matches: [],
};

export default function BottomNavigation() {
  const user = useSelector(selectUser);
  if (!user) return <></>;

  return (
    <div className={styles.container}>
      <Item icon={home} path="/" label="Home" matches={['/timer']} />
      <Item icon={plus} path="/new" label="New Task" />
      <Item icon={editTask} path="/edit" label="Edit Tasks" />
      <Item icon={chart} path="/charts" label="Charts" />
    </div>
  );
}
