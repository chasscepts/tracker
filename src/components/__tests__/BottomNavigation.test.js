import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Wrapper from '../../test_helpers/wrapper';
import BottomNavigation from '../../containers/BottomNavigation';
import { createStore } from '../../app/store';
import login from '../../test_helpers/login';

describe('BottomNavigation', () => {
  it('matches snapshot', async () => {
    const store = createStore();
    login(store);
    const tree = renderer
      .create(<Wrapper store={store} Component={BottomNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays all the links', () => {
    const store = createStore();
    login(store);
    const links = ['Home', 'New Task', 'Edit Tasks', 'Charts'];
    render(<Wrapper store={store} Component={BottomNavigation} />);
    links.forEach((link) => expect(screen.getByText(link)).toBeInTheDocument());
  });
});
