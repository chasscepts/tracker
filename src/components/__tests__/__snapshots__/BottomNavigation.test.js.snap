// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation matches snapshot 1`] = `
<div
  className="container"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="nav active"
    >
      <img
        alt="/"
        src="home.png"
      />
      <div
        className="label"
      >
        Home
      </div>
    </div>
  </a>
  <a
    href="/new"
    onClick={[Function]}
  >
    <div
      className="nav"
    >
      <img
        alt="/new"
        src="plus.png"
      />
      <div
        className="label"
      >
        New Task
      </div>
    </div>
  </a>
  <a
    href="/edit"
    onClick={[Function]}
  >
    <div
      className="nav"
    >
      <img
        alt="/edit"
        src="edit-task.png"
      />
      <div
        className="label"
      >
        Edit Tasks
      </div>
    </div>
  </a>
  <a
    href="/charts"
    onClick={[Function]}
  >
    <div
      className="nav"
    >
      <img
        alt="/charts"
        src="chart.png"
      />
      <div
        className="label"
      >
        Charts
      </div>
    </div>
  </a>
</div>
`;
